src/Razor/test/Microsoft.AspNetCore.Razor.Test.Common.Tooling/LanguageServer/TestRazorDocumentServiceProvider.cs

```diff
@@ -14,18 +14,23 @@ internal class TestRazorDocumentServiceProvider(IRazorSpanMappingService spanMap
 
     public bool SupportDiagnostics => true;
 
-    TService IRazorDocumentServiceProvider.GetService<TService>()
+    TService? IRazorDocumentServiceProvider.GetService<TService>() where TService : class
```

Comment:
I don't think this constraint is necessary, especially considering that we're checking against interfaces

Reply:
without it, I'm getting

![image](https://github.com/user-attachments/assets/3af517ee-3437-4cf0-9089-d7a556a3a49c)

Reply:
adding the constraint (like the interface has) gets rid of the error, which is why I added it there.